src/backend/commands/trigger.c

```diff
@@ -4899,13 +4903,18 @@ MakeTransitionCaptureState(TriggerDesc *trigdesc, Oid relid, CmdType cmdType)
 	 * delete statment as well, so we'll init both old and new transition tables
 	 * for Tsql dialect
 	 * */
-	if (sql_dialect == SQL_DIALECT_TSQL && (need_old || need_new))
+	if (sql_dialect == SQL_DIALECT_TSQL && (need_old_del || need_new_ins))
 	{
-		need_old = true;
-		need_new = true;
+		need_new_ins = true;
+		need_old_del = true;
+		/*
+		 * FIXME: setting need_new_upd to true seems to be working currently
+		 * but we need to investigate these flags and fix the implementation.
+		 */
+		need_new_upd = true;
 	}
```

Comment:
merge conflict resolution. We have filed https://jira.rds.a2z.com/browse/BABEL-5628 to fix this TODO